pkg/manager/server.go

```diff
@@ -21,41 +21,97 @@ import (
 	"errors"
 	"net"
 	"net/http"
+	"time"
 
 	"github.com/go-logr/logr"
+
+	crlog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var (
+	_ Runnable               = &Server{}
+	_ LeaderElectionRunnable = &Server{}
 )
 
-// server is a general purpose HTTP server Runnable for a manager
-// to serve some internal handlers such as health probes, metrics and profiling.
-type server struct {
-	Kind     string
-	Log      logr.Logger
-	Server   *http.Server
+// Server is a general purpose HTTP server Runnable for a manager.
+// It is used to serve some internal handlers for health probes and profiling,
+// but it can also be used to run custom servers.
+type Server struct {
+	// Kind is an optional string that describes the purpose of the server. It is used in logs to distinguish
+	// among multiple servers.
+	Kind string
+
+	// Log is the logger used by the server. If not set, a logger will be derived from the context passed to Start.
```

Comment:
Why not only have one way to pass it in?

Reply:
I didn't want to require a `Server` struct to be setup with a logger in advance. I don't have a strong opinion here though.

Reply:
Could we remove the Log here, and just inherit the logger from Start?

Reply:
This was already present in the unexported `server` struct, so I wasn't going to change it.

There is existing code in the pprof and health probe server setup that explicitly sets the servers' logger to the controller manager's logger (which comes from manager.Options.Logger). 

There's a bunch of context plumbing that makes it difficult to tell which contexts are used where, but I can't find anywhere that injects the controllerManager's logger into the context that is used to start the http servers. 

We could potentially make a custom context with the logger injected and use that context _just_ with the HTTPServers group to preserve existing behavior.

WDYT?